cassandra/components.py

```diff
@@ -650,67 +665,62 @@ def run_component(self):
         import xanthos
 
         config_file = self.params["config_file"]
-
         xth = xanthos.Xanthos(config_file)
 
         args = {}
+        gridded_runoff = []
+
+        # Other components should produce gridded climate data as a list of 2d numpy arrays
+        cap_names = ['gridded_pr', 'gridded_tas', 'gridded_pr_coord', 'gridded_tas_coord']
+        if all(cap in self.cap_tbl for cap in cap_names):
+            pr_grids = self.fetch('gridded_pr')
+            tas_grids = self.fetch('gridded_tas')
+            pr_coord = self.fetch('gridded_pr_coord')
+            tas_coord = self.fetch('gridded_tas_coord')
+        else:
+            pr_grids = tas_grids = []
+            xth_results = xth.execute(args)
+            gridded_runoff.append(xth_results.Q)
 
-        # Wait for other components to produce precipitation and
-        # temperature data (as pandas DataFrames), if provided
-        self.add_args_from_capability(args, 'PrecipitationFile', 'gridded_pr', 'mm_month-1')
-        self.add_args_from_capability(args, 'trn_tas', 'gridded_tas', 'C')
-
-        xth_results = xth.execute(args)
+        # Run Xanthos for each pair of precipitation and temperature grids
+        for pr, tas in zip(pr_grids, tas_grids):
```

Comment:
This loop can only ever be executed if the `true` branch of the conditional on line 675 executes, correct?  If so, would it make more sense to nest this loop inside that conditional to make that relationship a little more clear.

If we don't do that, then we probably need some additional comments, 
1. inside the `true` branch of the conditional, indicating that the Xanthos run will be deferred until further down.  (At first blush it looked to me like Xanthos doesn't actually get executed if the input capabilities are available.)
2. At the top of the for loop, reminding readers that `pr_grids` and `tas_grids` are always empty if Xanthos has already executed, and alway non-empty if it hasn't.  In other words, someone reading the code shouldn't have to reason out that Xanthos is always being executed the right number of times (which in fact it is).